src/main/java/by/triumgroup/recourse/service/impl/CourseServiceImpl.java

```diff
@@ -1,49 +1,95 @@
 package by.triumgroup.recourse.service.impl;
 
+import by.triumgroup.recourse.entity.dto.ErrorMessage;
 import by.triumgroup.recourse.entity.model.Course;
+import by.triumgroup.recourse.entity.model.User;
 import by.triumgroup.recourse.repository.CourseRepository;
+import by.triumgroup.recourse.repository.UserRepository;
 import by.triumgroup.recourse.service.CourseService;
+import by.triumgroup.recourse.validation.exception.ServiceBadRequestException;
+import by.triumgroup.recourse.validation.exception.ServiceNotFoundException;
 import org.springframework.data.domain.Pageable;
 import org.springframework.validation.Validator;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
 
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
 import static by.triumgroup.recourse.util.RepositoryCallWrapper.wrapJPACall;
+import static by.triumgroup.recourse.util.RepositoryCallWrapper.wrapJPACallToOptional;
 
+@SuppressWarnings("OptionalGetWithoutIsPresent")
 public class CourseServiceImpl
         extends AbstractCrudService<Course, Integer>
         implements CourseService {
 
-    private final CourseRepository repository;
+    private final CourseRepository courseRepository;
+    private final UserRepository userRepository;
 
-    public CourseServiceImpl(CourseRepository repository) {
-        super(repository);
-        this.repository = repository;
+    public CourseServiceImpl(CourseRepository courseRepository, UserRepository userRepository) {
+        super(courseRepository);
+        this.courseRepository = courseRepository;
+        this.userRepository = userRepository;
     }
 
     @Override
     public List<Course> searchByTitle(String title, Pageable pageable) {
-        return wrapJPACall(() -> repository.findByTitleContainingIgnoreCaseOrderByIdDesc(title, pageable));
+        return wrapJPACall(() -> courseRepository.findByTitleContainingIgnoreCaseOrderByIdDesc(title, pageable));
     }
 
     @Override
     public List<Course> findByStatus(Course.Status status, Pageable pageable) {
-        return wrapJPACall(() -> repository.findByStatusOrderByIdDesc(status, pageable));
+        return wrapJPACall(() -> courseRepository.findByStatusOrderByIdDesc(status, pageable));
     }
 
     @Override
-    public boolean registerStudentToCourse(Integer studentId, Course courseId) {
-//        Set<User> registeredStudents = course.getStudents();
-//        registeredStudents.add(student);
-//        wrapJPACall(() -> repository.save(course))
-        throw new NotImplementedException();
+    public void registerStudentToCourse(Integer studentId, Integer courseId) {
+        Optional<Course> courseOptional = wrapJPACallToOptional(() -> courseRepository.findOne(courseId));
+        Optional<User> user = wrapJPACallToOptional(() -> userRepository.findOne(studentId));
+        List<ErrorMessage> errorMessages = new ArrayList<>();
+        if (!courseOptional.isPresent()) {
```

Comment:
Consider refactor this if series. User else to show everyone, that you should continue checks, if a user and a course are found.